cogs/mod.py

```diff
@@ -204,7 +277,7 @@ def __init__(self, bot):
 
     @commands.command(no_pm=True, pass_context=True)
     @checks.admin_or_permissions(ban_members=True)
-    async def ban(self, ctx, user: discord.Member, days: int=0):
+    async def ban(self, ctx, user: discord.Member, days: int=0, *, reason: str = None):
```

Comment:
Check to see if the first word is a number, then have the caller confirm if it's part of the reason or the number of days.